### PR TITLE
global: Flask-Security refactoring

### PR DESCRIPTION
**Pull request by lnielsen**
* INCOMPATBILE Refactors code to use Flask-Security instead of
  custom implementation.

Signed-off-by: Lars Holm Nielsen <lars.holm.nielsen@cern.ch>

### CODE REVIEW

File: MANIFEST.in [hunk 1]
```diff
@@ -20,24 +22,35 @@
...
 recursive-include docs Makefile
-recursive-include invenio_accounts *.css
+recursive-include examples *.html
+recursive-include examples *.py
 recursive-include invenio_accounts *.html
-recursive-include invenio_accounts *.js
-recursive-include invenio_accounts *.py
-recursive-include invenio_accounts *.tpl
+recursive-include invenio_accounts *.po
+recursive-include misc *.py
```
Comment (jirikuncar) [inline MANIFEST.in]: can we remove the `misc` directory?

File: docs/usage.rst [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+    along with Invenio; if not, write to the
+    Free Software Foundation, Inc., 59 Temple Place, Suite 330, Boston,
+    MA 02111-1307, USA.
+
+    In applying this license, CERN does not
+    waive the privileges and immunities granted to it by virtue of its status
+    as an Intergovernmental Organization or submit itself to any jurisdiction.
+
+
+=======
```
Comment (jirikuncar) [inline docs/usage.rst]:
```rst
Usage
=====
```
?

File: invenio_accounts/cli.py [hunk 3]
```diff
@@ -0,0 +1,124 @@
...
+
+
+@users.command()
+@click.option('-u', '--user')
+@with_appcontext
+def deactivate(user):
+    """Deactivate a user."""
+    user_obj = _datastore.get_user(user)
+    if user_obj is None:
+        click.echo("ERROR: User not found.")
```
Comment (jirikuncar) [inline invenio_accounts/cli.py]: - `'` vs. `"` ?
- `click.secho(..., fg='red')`

File: invenio_accounts/templates/invenio_accounts/_macros.html [hunk 4]
```diff
@@ -0,0 +1,30 @@
+{% macro render_field(field, icon="", autofocus=False, errormsg=True) %}
```
Comment (jirikuncar) [inline invenio_accounts/templates/invenio_accounts/_macros.html]: licence header?

File: invenio_accounts/version.py [hunk 5]
```diff
@@ -28,4 +28,6 @@
 and parsed by ``setup.py``.
 """
 
-__version__ = "0.2.1.dev20151005"
+from __future__ import absolute_import, print_function
```
Comment (jirikuncar) [inline invenio_accounts/version.py]: IMHO we don't need it here ...

**Comment by jirikuncar** [comment]
@SamiHiltunen please have a look at new structure of Invenio package as proposed in the PR.

File: invenio_accounts/version.py (hunk 5 above)
Comment (lnielsen) [inline invenio_accounts/version.py]: It's to standardise every python file to always have the same ``__future__`` imports. Not something I feel strongly about, but we should fix it upstream in cookiecutter.

File: MANIFEST.in (hunk 1 above)
Comment (lnielsen) [inline MANIFEST.in]: Yes, I guess we can simply remove it in the cookie cutter post clean up script (inveniosoftware/cookiecutter-invenio-module#29)

**Comment by kaplun** [comment]
The change is huge. Are there some particular area you wish for even more review?

**Comment by lnielsen** [comment]
@kaplun Thanks. I think right now it's probably not necessary, as we anyway need some further iterations over it before it makes sense to review closer. I think by beginning of November we will have the first versions ready where it will make sense to review it. It's pretty much in line with what has been discussed previously of moving fully to Flask-Security. Also note the PR is for ``next`` branch, so won't affect ``master`` yet.

File: invenio_accounts/models.py [hunk 6]
```diff
@@ -1,235 +1,74 @@
...
+    id = db.Column(db.Integer(), primary_key=True)
+
+    name = db.Column(db.String(80), unique=True)
+
+    description = db.Column(db.String(255))
+
+
+class User(db.Model, UserMixin):
+    """User data model."""
+
```
Comment (jirikuncar) [inline invenio_accounts/models.py]: @lnielsen do we need to specify `__tablename__` for `User` and `Role`?

File: invenio_accounts/templates/invenio_accounts/base-cover.html [hunk 7]
```diff
@@ -1,4 +1,4 @@
-# -*- coding: utf-8 -*-
+{# -*- coding: utf-8 -*-
```
Comment (jirikuncar) [inline invenio_accounts/templates/invenio_accounts/base-cover.html]: I'm not sure about dash (`-`) in template name.

File: invenio_accounts/version.py (hunk 5 above)
Comment (jirikuncar) [inline invenio_accounts/version.py]: No problem. It is just that any of the directives is needed in this specific file.

File: tests/test_invenio_accounts.py [hunk 8]
```diff
@@ -0,0 +1,72 @@
...
+    ext.init_app(app)
+    assert 'invenio-accounts' in app.extensions
+
+
+def test_view(app):
+    """Test view."""
+    Babel(app)
+    InvenioAccounts(app)
+    with app.test_client() as client:
+        res = client.get("/login")
```
Comment (jirikuncar) [inline tests/test_invenio_accounts.py]: shall we try to keep compatibility at least for common urls (`/youraccount/login`)?

**Comment by jirikuncar** [comment]
@lnielsen It looks good. Please have a look at my comments above and you can add `Reviewed-by: me`.

File: tests/test_invenio_accounts.py (hunk 8 above)
Comment (lnielsen) [inline tests/test_invenio_accounts.py]: IMHO no :-) I never liked the "youraccount" and also you can configure the URL via variables.

File: invenio_accounts/templates/invenio_accounts/base-cover.html (hunk 7 above)
Comment (lnielsen) [inline invenio_accounts/templates/invenio_accounts/base-cover.html]: Agreed. Will change it.

File: tests/test_invenio_accounts.py (hunk 8 above)
Comment (jirikuncar) [inline tests/test_invenio_accounts.py]: :+1:

File: invenio_accounts/models.py (hunk 6 above)
Comment (lnielsen) [inline invenio_accounts/models.py]: Should we go with a naming scheme: ``accounts_role``? or will we hit the mysql 16 characters limit too fast?

Comment (jirikuncar) [inline invenio_accounts/models.py]: Personally, I don't care about MySQL so much as long as there is not ANSI limitation. So let's make understandable names.

File: docs/index.rst [hunk 9]
```diff
@@ -21,19 +21,43 @@
...
+
 .. include:: ../README.rst
 
-Contents:
-=========
+User's Guide
+------------
+
+This part of the documentation will show you how to get started in using
+Invenio-Base.
```
Comment (jirikuncar) [inline docs/index.rst]: `s/Invenio-Base/Invenio-Accounts/`

**Pull request merged**